modules/gapi/src/backends/common/serialization.cpp

```diff
@@ -361,6 +366,140 @@ I::IStream& operator>> (I::IStream& is, cv::GArg &arg) {
+    switch (arg.index()) {
+#if !defined(GAPI_STANDALONE)
+    case cv::GRunArg::index_of<cv::UMat>() :
+        os << cv::util::get<cv::UMat>(arg);       break;
+#endif
+    case cv::GRunArg::index_of<cv::Mat>() :
+        os << cv::util::get<cv::Mat>(arg);        break;
+    case cv::GRunArg::index_of<cv::Scalar>() :
+        os << cv::util::get<cv::Scalar>(arg);     break;
+    default: GAPI_Assert(false && "GRunArg: Unsupported type for <<");
+    }
+    return os;
+}
+I::IStream& operator >> (I::IStream& is, cv::GRunArg &arg) {
+    uint32_t index;
+    is >> index;
+#if !defined(GAPI_STANDALONE)
+    cv::UMat  v0;
+#endif
+    cv::Mat v1;
+    cv::Scalar v2;
+    switch (index) {
+#if !defined(GAPI_STANDALONE)
+    case cv::GRunArg::index_of<cv::UMat>() :
+        is >> v0; arg = v0; break;
+#endif
+    case cv::GRunArg::index_of<cv::Mat>() :
+        is >> v1; arg = v1; break;
+    case cv::GRunArg::index_of<cv::Scalar>() :
+        is >> v2; arg = v2; break;
+    default: GAPI_Assert(false && "GRunArg: Unsupported type for >>");
+    }
+    return is;
+}
+I::IStream& operator >> (I::IStream& is, cv::GRunArgP &arg) {
+    GRunArg res_obj;
+    is >> res_obj;
+    using T = cv::GRunArg;
+    switch (res_obj.index())
+    {
+#if !defined(GAPI_STANDALONE)
+    case T::index_of<cv::UMat>() :
+    {
+        *cv::util::get<cv::UMat*>(arg) = cv::util::get<cv::UMat>(res_obj);
+    } break;
+#endif
+    case T::index_of<cv::Mat>() :
+    {
+        *cv::util::get<cv::Mat*>(arg) = cv::util::get<cv::Mat>(res_obj);
+    } break;
+    case T::index_of<cv::Scalar>() :
+        *cv::util::get<cv::Scalar*>(arg) = cv::util::get<cv::Scalar>(res_obj);
+        break;
+    default:
+        GAPI_Assert(false && "This value type is not supported!"); // ...maybe because of STANDALONE mode.
+        break;
+    }
+    return is;
+}
+
+GAPI_EXPORTS void bind(cv::GRunArg &res_obj, cv::GRunArgP &out_obj)
+{
+    // FIXME: this conversion should be unified
+    using T = cv::GRunArgP;
+    switch (out_obj.index())
+    {
+#if !defined(GAPI_STANDALONE)
+    case T::index_of<cv::UMat*>() :
+    {
+        *cv::util::get<cv::UMat*>(out_obj) = cv::util::get<cv::UMat>(res_obj);
+    }
+    break;
+#endif
+    case T::index_of<cv::Mat*>() :
+    {
+        *cv::util::get<cv::Mat*>(out_obj) = cv::util::get<cv::Mat>(res_obj);
+    } break;
+    case T::index_of<cv::Scalar*>() :
+        *cv::util::get<cv::Scalar*>(out_obj) = cv::util::get<cv::Scalar>(res_obj);
+        break;
+    case T::index_of<cv::detail::VectorRef>() :
+        cv::util::get<cv::detail::VectorRef>(out_obj).mov(cv::util::get<cv::detail::VectorRef>(res_obj));
+        break;
+    case T::index_of<cv::detail::OpaqueRef>() :
+        cv::util::get<cv::detail::OpaqueRef>(out_obj).mov(cv::util::get<cv::detail::OpaqueRef>(res_obj));
+        break;
+    default:
+        GAPI_Assert(false && "This value type is not supported!"); // ...maybe because of STANDALONE mode.
+        break;
+    }
+}
```

Comment:
The idea of `bind` is to produce a vector-of-pointers for a vector-of-objects.
You have a vector of objects, and all you need is to create a vector of pointers to that objects.
Here you makes copy which is not actually required. We just need to take pointers and that's it.

Reply:
Done. Refactored, but still not sure that it's correct implementation.